components/aero_sensors/HW/HW_gpio.c

```diff
@@ -0,0 +1,111 @@
+ * @author Joshua Lafleur (josh.lafleur@outlook.com)
+ * @version 0.1
+ * @date 2022-07-02
+ */
+
+/******************************************************************************
+ *                             I N C L U D E S
+ ******************************************************************************/
+
+#include "SystemConfig.h"
+
+#include "HW_gpio.h"
+
+
+/******************************************************************************
+ *                       P U B L I C  F U N C T I O N S
+ ******************************************************************************/
+
+/**
+ * @brief  Initializes the IO pins of the component
+ */
+void HW_GPIO_Init(void)
+{
+    GPIO_InitTypeDef GPIO_InitStruct = { 0 };
+
+    // Enable GPIO clocks
+    __HAL_RCC_GPIOA_CLK_ENABLE();
+    __HAL_RCC_GPIOB_CLK_ENABLE();
+    __HAL_RCC_GPIOC_CLK_ENABLE();
+    __HAL_RCC_GPIOD_CLK_ENABLE();
+    __HAL_RCC_AFIO_CLK_ENABLE();
+
+    // Turn Board LED off
+    HAL_GPIO_WritePin(LED_GPIO_Port, LED_Pin, GPIO_PIN_RESET);
+    // Configure Board LED pin
+    GPIO_InitStruct.Pin   = LED_Pin;
+    GPIO_InitStruct.Mode  = GPIO_MODE_OUTPUT_PP;
+    GPIO_InitStruct.Pull  = GPIO_PULLDOWN;
+    GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_LOW;
+    HAL_GPIO_Init(LED_GPIO_Port, &GPIO_InitStruct);
+
+    // Turn Status LED off
+    HAL_GPIO_WritePin(LED_R_Port, LED_R_Pin, GPIO_PIN_RESET);
+    HAL_GPIO_WritePin(LED_G_Port, LED_G_Pin, GPIO_PIN_RESET);
+    HAL_GPIO_WritePin(LED_B_Port, LED_B_Pin, GPIO_PIN_RESET);
+    // Configure Status LED pin
+    GPIO_InitStruct.Pin   = LED_R_Pin | LED_G_Pin | LED_B_Pin;
+    GPIO_InitStruct.Mode  = GPIO_MODE_OUTPUT_PP;
+    GPIO_InitStruct.Pull  = GPIO_PULLDOWN;
+    GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_LOW;
+    HAL_GPIO_Init(LED_GPIO_Port, &GPIO_InitStruct);
+    
+    /**< Initialize I2C1 */
+    AFIO->MAPR |= 0 << AFIO_MAPR_I2C1_REMAP_Pos; /**< No remap */
+
+    GPIO_InitStruct.Pin = I2C1_SCL_Pin | I2C1_SDA_Pin;
+    GPIO_InitStruct.Mode = GPIO_MODE_AF_OD;
+    GPIO_InitStruct.Pull = GPIO_PULLUP;
+    GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_HIGH;
+    HAL_GPIO_Init(I2C1_SCL_Port, &GPIO_InitStruct);
+    
+    /**< Initialize I2C2 */
+    GPIO_InitStruct.Pin = I2C2_SCL_Pin | I2C2_SDA_Pin;
+    GPIO_InitStruct.Mode = GPIO_MODE_AF_OD;
+    GPIO_InitStruct.Pull = GPIO_PULLUP;
+    GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_HIGH;
+    HAL_GPIO_Init(I2C2_SCL_Port, &GPIO_InitStruct);
+}
+
+/**
+ * @brief Reads the input pin state
+ * @param[in] input Input pin
+ * @return HW pin state
+ */
+HW_GPIO_PinState_E HW_GPIO_ReadPin(const HW_GPIO_S *input)
+{
+    GPIO_PinState bitstatus;
+
+    /* Check the parameters */
+    assert_param(IS_GPIO_PIN(input->pin));
+
+    if ((input->port->IDR & input->pin) != (uint32_t)GPIO_PIN_RESET)
+    {
+        bitstatus = GPIO_PIN_SET;
+    }
+    else
+    {
+        bitstatus = GPIO_PIN_RESET;
+    }
+    return (HW_GPIO_PinState_E)bitstatus;
+}
```

Comment:
Why are you making a `GPIO_PinState` type variable just to cast it to `HW_GPIO_PinState_E` when returning? What is the point of having `GPIO_PinState` at all?

```suggestion
typedef enum
{
    GPIO_PIN_RESET = 0U,
    GPIO_PIN_SET,
} GpioState_T;

GpioState_T GPIO_readPin(const HW_GPIO_S *input)
{
    GpioState_T bitStatus = GPIO_PIN_RESET;

    if ((input->port->IDR & input->pin) != (uint32_t)GPIO_PIN_RESET)
    {
        bitStatus = GPIO_PIN_SET;
    }
    
    return bitStatus;
}
```

I also don't like `HW_GPIO_S` but can't think of anything better right now